bundler/lib/bundler/ui/shell.rb

```diff
@@ -119,6 +128,8 @@ def unprinted_warnings
 
       # valimism
       def tell_me(msg, color = nil, newline = nil)
+        return tell_err(msg, color, newline) unless interactive?
```

Comment:
This is the core of the PR: if we are in non-interactive mode, route debug/info/confirm messages to stderr instead of stdout.

Reply:
Maybe the "interactive" name is a bit confusing here with how "interactive" is usually used in the context of shells? For example, if running in CI, the shell will be non-interactive in the sense that there's no prompt available to provide input, but this option will still be true for `bundle lock`.

Maybe a `stream_for_progress` accessor that defaults to stdout or something like that?

Reply:
I agree "interactive" is a loaded word; I struggled with the naming.

Maybe something like:

```ruby
ui.stream_for_progress = :stderr if print # defaults to :stdout
```

I hesitate to use the word `progress` because the stream is used for `info` and `confirm` messages, which by convention are for progress, but we don't actually use that terminology anywhere.

Alternatively:

```ruby
ui.output_stream = :stderr
```

I don't really have a strong opinion on this, though, so I am happy to go with whichever name you feel is more intuitive.

Or are you suggesting passing the stream itself, instead of a symbol, like this?

```ruby
ui.stream_for_progress = $stderr
```

That might be tricky, because I think `Thor::Shell::Basic` doesn't have an accessor to change the stream it uses.

Reply:
I think `ui.output_stream = :stderr` is good 👍.

Reply:
Done! ✔️